### ISSUE TITLE
[Bug] Error while saving Spam Data

### ISSUE THREAD

**Issue by KingDGrizzle** [GilbertGobbels/GAwesomeBot#31]
Welp, found the part of the code that errors, now to debug why it errors.

`setTimeout(() => {
							spamDocument = channelDocument.spam_filter_data.id(msg.author.id);
							if(spamDocument) {
								spamDocument.remove();
								serverDocument.save(err => {
									winston.error("Failed to save server data for spam filter", {svrid: msg.guild.id}, err);
								});
							}
						}, 45000);`

**Comment by KingDGrizzle** [comment]
`if(serverDocument.config.moderation.isEnabled && serverDocument.config.moderation.filters.spam_filter.isEnabled && serverDocument.config.moderation.filters.spam_filter.disabled_channel_ids.indexOf(msg.channel.id)==-1 && memberBotAdmin<1) {
					// Tracks spam with each new message (auto-delete after 45 seconds)
					let spamDocument = channelDocument.spam_filter_data.id(msg.author.id);
					if(!spamDocument) {
						channelDocument.spam_filter_data.push({_id: msg.author.id});
						spamDocument = channelDocument.spam_filter_data.id(msg.author.id);
						spamDocument.message_count++;
						spamDocument.last_message_content = msg.cleanContent;
						setTimeout(() => {
							spamDocument = channelDocument.spam_filter_data.id(msg.author.id);
							if(spamDocument) {
								spamDocument.remove();
								serverDocument.save(err => {
									winston.error("Failed to save server data for spam filter", {svrid: msg.guild.id}, err);
								});
							}
						}, 45000);
					// Add this message to spamDocument if similar to the last one
					}`
Something in HERE breaks, which is weard and off

**Comment by KingDGrizzle** [comment]
And while its obvious that winston will always output, I did a quick if check and it still outputs.

**Comment by HilbertGilbertson** [comment]
Just looked at this. Had I not been reading through Eris source the other day I wouldn't have spotted it. `{svrid: msg.guild.id}` should be `{svrid: msg.channel.guild.id}` as msg.guild produces a deprecation warning.

**Comment by HilbertGilbertson** [comment]
The above being said, while spam data now saves to the db, there seem to be other flaws in spam handling, as it just ends up looking like:

```
"spam_filter_data": [
        {
          "last_message_content": "test",
          "_id": "225749221188435969",
          "message_count": NumberInt(1)
        },
        {
          "last_message_content": "test",
          "_id": "225749221188435969",
          "message_count": NumberInt(1)
        },
        {
          "last_message_content": "test",
          "_id": "225749221188435969",
          "message_count": NumberInt(1)
        },
        {
          "last_message_content": "test",
          "_id": "225749221188435969",
          "message_count": NumberInt(1)
        }
      ]
```

i.e. the message count isn't increasing and no warnings are sent to the user/admins, and no actions are triggered.

**Comment by KingDGrizzle** [comment]
Then we should re-write the whole spamDocument part. it should check for the id and if it finds it add +1 to count (if its the same)

**Comment by HilbertGilbertson** [comment]
@rgiyf Has helpfully proposed (PR #91) updating all instances of `msg.guild` to `msg.channel.guild`, but as I've noted above, please remember this does not fix the underlying problems in spam handling.

**Comment by HilbertGilbertson** [comment]
Fixed in #102

**Comment by KingDGrizzle** [comment]
Closed as the Spam Filter works #102
